rocket_twin/drivers/fueling_rocket.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+
+from cosapp.drivers import Driver, NonLinearSolver, RungeKutta
+from cosapp.recorders import DataFrameRecorder
+from cosapp.systems import System
+
+
+class FuelingRocket(Driver):
+    """Driver that simulates the fueling of a rocket.
+
+    Inputs
+    ------
+    name: string,
+        the name of the driver
+    w_out [kg/s]: float,
+        mass flow of fuel exiting the ground tank
+    dt [s]: float,
+        integration time step
+    owner: System,
+        the system that owns the driver
+
+    Outputs
+    ------
+    """
+
+    def __init__(self, name: str, w_out, dt, owner: Optional["System"] = None, **kwargs):
+        super().__init__(name, owner, **kwargs)
+
+        # Fueling:
+        self.rk = self.add_driver(RungeKutta("rk", owner=owner, order=4, dt=dt))
+        self.rk.time_interval = (0.0, 1.0)
+        self.solver = self.rk.add_child(NonLinearSolver("solver"))
+        self.time_end = 0.0
+
+        init = {
+            "rocket.dyn.switch": False,
+            "rocket.tank.w_out_temp": 0.0,
+            "pipe.is_open": True,
+            "g_tank.w_in": 0.0,
+            "g_tank.w_out_temp": w_out,
+        }
+
```

Comment:
could be removed if init and stop are in the input variables

Reply:
Check